pkg/controller/job/job_controller_test.go

```diff
@@ -725,14 +721,13 @@ func TestControllerSyncJob(t *testing.T) {
 	}
 
 	for name, tc := range testCases {
-		for _, wFinalizers := range []bool{false, true} {
-			t.Run(fmt.Sprintf("%s, finalizers=%t", name, wFinalizers), func(t *testing.T) {
+		// NOTE we should remove this but it makes the git diffs really ugly right now.
+		// Leave for early review process.
+		for _, wFinalizers := range []bool{true} {
+			t.Run(fmt.Sprintf("%s", name), func(t *testing.T) {
 				if wFinalizers && tc.podControllerError != nil {
 					t.Skip("Can't track status if finalizers can't be removed")
 				}
```

Comment:
We should remove this check and update the test cases.

When there are pod update failures, the status should simply stay the same.

Reply:
Should we do that as part of this PR?  Your request seems to be a little outside of the legacy tracking removal?  But we could do it as a separate issue so we keep PR smaller?

Reply:
I'm trying to create a ticket for this but I don't really understand the behavior you want.  I think the comment is wrong.  

I see us setting podControllerError when we are testing Delete/Create in the SyncJob and we are getting an error when attempting to create/delete pods.  What is your intention with removing this?

Reply:
Note that this error is used for any pod action: create or update.

In the legacy behavior, if there was a pod create failure, the controller would still update the Job status with any pod terminations it sees.
In the new behavior, it can't update the status, because it needs to update the finalizers first. So that's why I disabled those tests. Now that the legacy behavior is gone, we can fix the tests to make sure that there was no Job status update on the succeeded/failed counters.

Reply:
Okay.  After this PR gets merged, I'll open a PR with updating these test cases.